test/unit/ipam_test.rb

```diff
@@ -45,6 +45,49 @@ class IPAMTest < ActiveSupport::TestCase
     end
   end
 
+  context 'random db' do
+    test "should find unused IPv4" do
+      subnet = FactoryGirl.create(
+        :subnet_ipv4, :name => 'my_subnet',
+        :network => '10.0.0.0',
+        :mask => '255.0.0.0',
+        :ipam => IPAM::MODES[:random_db])
+      ipam = IPAM::RandomDb.new(:subnet => subnet)
+      assert_match /^10\./, ipam.suggest_ip
+    end
+
+    test "should return IPv4 based on MAC if provided" do
+      subnet = FactoryGirl.create(
+        :subnet_ipv4, :name => 'my_subnet',
+        :network => '10.0.0.0',
+        :mask => '255.0.0.0',
+        :ipam => IPAM::MODES[:random_db])
+      ipam1 = IPAM::RandomDb.new(:subnet => subnet, :mac => "AA:BB:CC:DD:EE:FF")
+      ipam2 = IPAM::RandomDb.new(:subnet => subnet, :mac => "AA:BB:CC:DD:EE:FF")
+      assert_equal ipam1.suggest_ip, ipam2.suggest_ip
+    end
+
+    test "should find unused IPv6" do
```

Comment:
If you want to keep that test, you also need to extend `supported_ipam_modes` for IPv6.